File: js/targetNumber.js
function solution(numbers, target) {
    var answer = 0;
    
    function recursive(i, sum){
        console.log("i : "+i + " sum : "+sum);
        if(i === numbers.length){
            if(sum === target){
                answer++;
            }
            return;
        }
        recursive(i+1,sum+numbers[i]);
        recursive(i+1,sum-numbers[i]);
    }
    recursive(0,0);
    return answer;
}

solution([1, 1, 1, 1, 1],3);

File: targetNumber/TargetNumber.java
package targetNumber;

public class TargetNumber {
    int count = 0;
    public int solution(int[] numbers, int target) {
        int answer = 0;
        
        answer += dfs(numbers,0,0,target);        
        return answer;
    }

    private int dfs(int[] numbers, int node, int sum, int target){
        count++;
        System.out.println(count);
        if(node == numbers.length){
            if(sum == target){
                return 1;
            }
            return 0;
        }
        return dfs(numbers, node+1, sum + numbers[node], target) + dfs(numbers, node+1, sum - numbers[node], target);
    }
    public static void main(String[] args) {
        int[] numbers = {1, 1, 1, 1, 1};
        new TargetNumber().solution(numbers, 3);
    }
}